cmake/api_macros.cmake

```diff
@@ -191,6 +191,14 @@ macro(OPENDDS_TARGET_SOURCES target)
     list(APPEND _tao_options "-SS")
   endif()
 
+  if(NOT ${OPENDDS_COMPILER} STREQUAL "cl" AND ${OPENDDS_HOST} STREQUAL "win32")
```

Comment:
Why is this conditional?  CMake always provides `CMAKE_CXX_COMPILER` and I don't know why we wouldn't just use it.

Reply:
I agree. You can also get rid of the configure script change.

Reply:
You can't get rid of the configure script change because it would result in CL being put into CMAKE_CXX_COMPILER with no path.